src/main/kotlin/net/atos/zac/app/besluit/BesluitService.kt

```diff
@@ -53,21 +139,24 @@ class BesluitService @Inject constructor(
                 .filter { it.informatieobject.extractUuid() == teVerwijderenInformatieobject }
                 .forEach { brcClientService.deleteBesluitinformatieobject(it.url.extractUuid()) }
         }
-        documentUuidsToAdd.forEach { documentUri ->
-            drcClientService.readEnkelvoudigInformatieobject(documentUri).let { enkelvoudigInformatieObject ->
+        documentUuidsToAdd.forEach { documentUUID ->
+            drcClientService.readEnkelvoudigInformatieobject(documentUUID).let { enkelvoudigInformatieObject ->
                 BesluitInformatieObject().apply {
                     this.informatieobject = enkelvoudigInformatieObject.url
                     this.besluit = besluit.url
                 }
             }.let {
-                brcClientService.createBesluitInformatieobject(
-                    it,
-                    WIJZIGEN_BESLUIT_TOELICHTING
-                )
+                brcClientService.createBesluitInformatieobject(it, WIJZIGEN_BESLUIT_TOELICHTING)
             }
         }
     }
 
+    fun withdrawBesluit(besluit: Besluit, reden: String): Besluit =
+        brcClientService.updateBesluit(
+            besluit,
+            getBesluitWithdrawalExplanation(besluit.vervalreden)?.let { String.format(it, reden) }
```

Comment:
Is there a max length on this field?

Reply:
would be nice if this is a builder instead... or just takes the Enum and a reason as parameters to build the explanation result string.